src/vs/platform/update/electron-main/updateMetadataProvider.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// {{SQL CARBON EDIT}} - This file contains interfaces, util functions and url definitions
+// to support Download Center based update.
+import { IProductService } from 'vs/platform/product/common/productService';
+import { IUpdate } from 'vs/platform/update/common/update';
+
+export const updateMetadataUrl = 'https://go.microsoft.com/fwlink/?linkid=2274438';
+export const updateMetadataMacUrl = 'https://go.microsoft.com/fwlink/?linkid=2274285';
+export const updateMetadataMacArmUrl = 'https://go.microsoft.com/fwlink/?linkid=2274463';
+export const updateMetadataMacUniversalUrl = 'https://go.microsoft.com/fwlink/?linkid=2274286';
+
+export interface Asset {
+	platform: string;
+	type: string;
+	url: string;
+	mooncakeUrl: string;
+	hash: string;
+	sha256hash?: string;
+	packageCatalog?: string;
+	repoDefinition?: string;
+	repoDataFiles?: string[];
+}
+
+export interface Build {
+	id: string;
+	version: string;
+	isFrozen?: boolean;
+	assets: Asset[];
+	updates: { [platform: string]: string; };
+}
+
+export function getUpdateFromBuild(build: Build | null, productService: IProductService, platform: string): IUpdate | undefined {
```

Comment:
This logic is copied from the update server. We are basically moving metadata parsing logic to the client side here.